python/perf-kernels/flash-attention.py

```diff
@@ -26,53 +28,87 @@
 import triton.language as tl
 
 torch_dtype:tl.constexpr = torch.float16
+
 TORCH_HAS_FP8E5 = hasattr(torch, 'float8_e5m2fnuz')
 if TORCH_HAS_FP8E5:
     torch_dtype:tl.constexpr = torch.float8_e5m2fnuz
 
-def prepare_bias(bias, batch, nheads, seqlen):
+@triton.jit
+def max_fn(x, y):
+    return tl.math.max(x, y)
+
+@triton.jit
+def dropout_offsets(philox_seed, philox_offset, dropout_p, m, n, stride):
+    ms = tl.arange(0, m)
+    ns = tl.arange(0, n)
+    return philox_offset + ms[:, None] * stride + ns[None, :]
+
+@triton.jit
+def dropout_rng(philox_seed, philox_offset, dropout_p, m, n, stride):
+    rng_offsets = dropout_offsets(philox_seed, philox_offset, dropout_p, m, n, stride).to(tl.uint32)
+    # TODO: use tl.randint for better performance
+    return tl.rand(philox_seed, rng_offsets)
+
+@triton.jit
+def dropout_mask(philox_seed, philox_offset, dropout_p, m, n, stride):
+    rng_output = dropout_rng(philox_seed, philox_offset, dropout_p, m, n, stride)
+    rng_keep = rng_output > dropout_p
+    return rng_keep
+
+# Only needed for testing
+def prepare_bias(bias, batch, nheads, seqlen_q, seqlen_k):
     assert bias.is_cuda
     assert bias.dim() == 4
-    if bias.shape[2:] == (1, seqlen):
+    if bias.shape[2:] == (1, seqlen_k):
         bias_type = "vector"
-    elif bias.shape[2:] == (seqlen, seqlen):
+    elif bias.shape[2:] == (seqlen_q, seqlen_k):
         bias_type = "matrix"
     else:
         raise RuntimeError(
             "Last 2 dimensions of bias must be (1, seqlen)" " or (seqlen, seqlen)"
         )
-    return bias.expand(batch, nheads, seqlen, seqlen), bias_type
+    return bias.expand(batch, nheads, seqlen_q, seqlen_k), bias_type
 
 @triton.jit
 def _attn_fwd_inner(
     acc, l_i, m_i, q,
     K_block_ptr, V_block_ptr,
     start_m,
+    seqlen_q,
```

Comment:
This doesn't seem like it'd be used since in here, everything is about k/v

Reply:
Noted, will verify.